plugins/flatpak/gs-flatpak.c

```diff
@@ -3666,6 +3667,51 @@ app_has_local_source (GsApp *app)
 		(url != NULL && g_str_has_prefix (url, "file://"));
 }
 
+static gboolean
+gs_flatpak_allow_install_or_update (GsFlatpak *self,GsApp *app, GsAppList *list, gboolean is_update)
+{
+	g_autoptr (GFile) installation_dir = NULL;
+	g_autofree gchar *path = NULL;
+	GsApp *runtime, *update_runtime;
+	guint64 runtime_delta_size = 0;
+	guint64 min_free_space = 0;
+	guint i;
+	struct statvfs buf;
+
+	installation_dir = flatpak_installation_get_path (self->installation);
+	path = g_file_get_path (installation_dir);
+
+	if (statvfs (path, &buf) == -1) {
+		g_debug ("statvfs: Couldn't query disk space information");
+		return FALSE;
+	}
+
+	runtime = gs_app_get_runtime (app);
+	update_runtime = gs_app_get_update_runtime (app);
+	if (is_update && runtime != update_runtime) {
+
+		/* Correctly setup the runtime sizes if they are updatable */
+		guint64 runtime_size = gs_app_get_size_installed (runtime);
+		guint64 update_runtime_size = gs_app_get_size_installed (update_runtime);
+
+		if ((runtime_delta_size = update_runtime_size - runtime_size) > update_runtime_size)
+			runtime_delta_size = runtime_size - update_runtime_size;
+
+		gs_app_list_remove (list, update_runtime);
+	}
+
+	min_free_space = runtime_delta_size;
+	for (i = 0; i < gs_app_list_length (list); i++) {
+		GsApp *app_temp = gs_app_list_index (list, i);
+		min_free_space += gs_app_get_size_installed (app_temp);
+	}
+
+	if (buf.f_bfree < (min_free_space / buf.f_bsize + 1))
```

Comment:
Why `+ 1` here? If it's to avoid that `f_bsize` may be zero (unlikely, right?), then you need to add parenthesis to it, as division takes precedence, and write a comment about why the `+ 1` is used.
However, I don't think `f_bsize` should be zero (but you should check it out).
Independently, I think maybe multiplying `f_bfree * f_bsize` makes it clearer.

Reply:
> Why + 1 here? If it's to avoid that f_bsize may be zero (unlikely, right?), then you need to add parenthesis to it, as division takes precedence, and write a comment about why the + 1 is used.
> However, I don't think f_bsize should be zero (but you should check it out).

Yes, the `f_bsize`=0 is highliy unlikely and I wasn't trying to add `1` to `f_bsize` either.
Earlier I was using `<=` so I added `+1` to make sure that we compare with one extra block. But I remember that I shifted from `<=` to `<` but I think forgot to remove `+1`. I'll remove it now.